zboxcore/sdk/renameworker.go

```diff
@@ -42,51 +42,82 @@ func (req *RenameRequest) getObjectTreeFromBlobber(blobber *blockchain.StorageNo
 	return getObjectTreeFromBlobber(req.ctx, req.allocationID, req.allocationTx, req.remotefilepath, blobber)
 }
 
-func (req *RenameRequest) renameBlobberObject(blobber *blockchain.StorageNode, blobberIdx int) (fileref.RefEntity, error) {
-	refEntity, err := req.getObjectTreeFromBlobber(req.blobbers[blobberIdx])
+func (req *RenameRequest) renameBlobberObject(
+	blobber *blockchain.StorageNode, blobberIdx int) (refEntity fileref.RefEntity, err error) {
+
+	defer func() {
+		if err != nil {
+			req.maskMU.Lock()
+			req.renameMask = req.renameMask.And(zboxutil.NewUint128(1).Lsh(uint64(blobberIdx)).Not())
+			req.maskMU.Unlock()
+		}
+	}()
+
+	refEntity, err = req.getObjectTreeFromBlobber(req.blobbers[blobberIdx])
 	if err != nil {
 		return nil, err
 	}
 
-	body := new(bytes.Buffer)
-	formWriter := multipart.NewWriter(body)
+	var resp *http.Response
+	for i := 0; i < 3; i++ {
+		if resp != nil {
+			resp.Body.Close()
+		}
 
-	_ = formWriter.WriteField("connection_id", req.connectionID)
-	formWriter.WriteField("path", req.remotefilepath)
-	formWriter.WriteField("new_name", req.newName)
+		body := new(bytes.Buffer)
+		formWriter := multipart.NewWriter(body)
 
-	formWriter.Close()
-	httpreq, err := zboxutil.NewRenameRequest(blobber.Baseurl, req.allocationTx, body)
-	if err != nil {
-		l.Logger.Error(blobber.Baseurl, "Error creating rename request", err)
-		return nil, err
-	}
-	httpreq.Header.Add("Content-Type", formWriter.FormDataContentType())
-	ctx, cncl := context.WithTimeout(req.ctx, (time.Second * 30))
-	err = zboxutil.HttpDo(ctx, cncl, httpreq, func(resp *http.Response, err error) error {
+		formWriter.WriteField("connection_id", req.connectionID)
+		formWriter.WriteField("path", req.remotefilepath)
+		formWriter.WriteField("new_name", req.newName)
+		formWriter.Close()
+
+		var httpreq *http.Request
+		httpreq, err = zboxutil.NewRenameRequest(blobber.Baseurl, req.allocationTx, body)
 		if err != nil {
-			l.Logger.Error("Rename : ", err)
-			return err
+			l.Logger.Error(blobber.Baseurl, "Error creating rename request", err)
+			return nil, err
+		}
+		httpreq.Header.Add("Content-Type", formWriter.FormDataContentType())
+		ctx, cncl := context.WithTimeout(req.ctx, time.Minute)
+		resp, err = zboxutil.Client.Do(httpreq.WithContext(ctx))
+		cncl()
+
+		if err != nil {
+			logger.Logger.Error("Rename: ", err)
+			return nil, err
 		}
 		defer resp.Body.Close()
+
+		var respBody []byte
+		respBody, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
```

Comment:
If http.StatusOK does not care about the resp data, this reading maybe could be avoided or maybe lead to unexpected error if the body is empty?